haystack/components/agents/agent.py

```diff
@@ -283,8 +285,26 @@ def _initialize_fresh_execution(
             init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=requires_async
         )
 
-        tool_invoker_inputs: dict[str, Any] = {}
-        generator_inputs: dict[str, Any] = {"tools": self.tools}
+        selected_tools: list[Tool] | Toolset = self.tools
```

Comment:
We aren't requiring python 3.10 yet for Haystack so we are still using `Union` by default. 
```suggestion
        selected_tools: Union[list[Tool], Toolset] = self.tools
```